cookingrecipe/src/main/java/com/cookingrecipe/service/recipe/RecipeServiceImp.java

```diff
@@ -125,4 +125,23 @@ public Recipe getRecipeById(Integer id) {
 		}
 		return responses;
 	}
+
+	@Override
+	public List<RecipeResponse> searchByName(String name) {
+		List<RecipeResponse> responses = new ArrayList<>();
+		try {
+			List<Recipe> recipes = recipeDAO.searchByName(name);
+			for(Recipe recipe:recipes) {
+				RecipeResponse recipeResponse = new RecipeResponse();
+				BeanUtils.copyProperties(recipe, recipeResponse);
+				responses.add(recipeResponse);
+			}
+			return responses;
+			
+		} catch (Exception e) {
+			e.printStackTrace();
+			throw e;
```

Comment:
search thì ko throw bé nè, ngoài ra, nếu có exception thì nên ghi log nhé, chứ chỉ in ra console thì nếu muốn điều trả lại sau 3-4 ngày là ko dc nè

Reply:
vâng